app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  after_save :update_posts_counter
+  def update_posts_counter
+    user.increment!(:posts_counter)
+  end
+
+  def recent_comment
+    Comment.order(created_at: :desc).limit(5)
```

Comment:
- [ ] We want to fetch the comments related to this post, not all comments. Kindly change to this:

  ```
  comments.order(created_at: :desc).limit(5)
  ```